File: Dsa/maxofSum.py
def maxSum(nums, k):
    maxVarriable=0
    maxSum=0
    for i in range(k):
        maxVarriable+=nums[i]
    maxSum=maxVarriable
    for i in range(k, len(nums)):
        maxVarriable+=nums[i]
        maxVarriable-=nums[i-k]

        maxSum=max(maxSum, maxVarriable)
    return maxSum

def CountWord(str):
    str=str.lower().split()
    freq={}
    for i in str:
        if i in freq:
            freq[i]+=1
        else:
            freq[i]=1
    return freq

def CountlwtterDigit(str):
    digit=0
    letter=0
    for i in str:
        if "0"<=i<="9":
            digit+=1
        if "a"<=i<="z":
            letter+=1
    return digit, letter
        
print(maxSum([2,3,4,5,3,3,5,2], 3))
print(CountWord("this is this is"))
print(CountlwtterDigit("dhbahjgew7385rrhwejf34"))

File: Dsa/common/maxofSum.js
function maxSumArr(nums, k) {
    let maxVariable = 0
    let maxSum = 0
    for (let i = 0; i < k; i++) {
        maxVariable += nums[i]
    }
    maxSum = maxVariable
    for (let i = k; i < nums.length; i++) {
        maxVariable += nums[i]
        maxVariable -= nums[i - k]

        maxSum = Math.max(maxSum, maxVariable)
    }
    return maxSum

}

function CountWord(str) {
    str = str.toLowerCase().split(" ")
    let freq = {}
    for (let i = 0; i < str.length; i++) {
        let ch = str[i]

        freq[ch] = (freq[ch] || 0) + 1

    }
    return freq
}

function CountLetterDigit(str){
    let letter=0
    let digit=0
    for(let i=0; i<str.length; i++){
        let ch=str[i]
        if(ch>="0" && ch<="9"){
            digit++
        }
        if(ch>="a" && ch<="z"){
            letter++
        }
    }
    return {letter, digit}
}

console.log(CountLetterDigit("872349hwedh8ey2hkjad89324ehdsiiay"))

console.log(CountWord("this is a fuck boy who fuck this is"))
console.log(maxSumArr([1, 2, 3, 4, 5, 6, 7], 3))